packages/docusaurus-theme-classic/src/theme/ScrollMonitorProvider/index.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useMemo, useRef} from 'react';
+
+import ScrollMonitorContext from '@theme/ScrollMonitorContext';
+import type {Props} from '@theme/ScrollMonitorProvider';
+
+function ScrollMonitorProvider(props: Props): JSX.Element {
```

Comment:
I don't have a good intuition for what a "scroll monitor" is.

Maybe at the end of the review it will click but for now I don't understand. Is there a better possible name?

Reply:
The lengthier description is "a flag for whether scroll events in `useScrollPosition` are enabled, and handlers to control that flag"

Have you got any ideas for a good name here? I agree that 'scroll monitor' does not intuitively explain the above